pkg/config/templateComponent.go

```diff
@@ -177,6 +178,32 @@ func (t *TemplateComponent) applyTemplate(tmplText string, userdata map[string]a
 	return b.String(), nil
 }
 
+func (t *TemplateComponent) applyTemplate(tmplVal any, userdata map[string]any) (any, error) {
+	switch k := tmplVal.(type) {
+	case string:
+		if tmplVal == "" || !strings.Contains(k, "{{") {
+			return k, nil
+		}
+		// the content of the template is a single value
+		// if we're just including a specific value, don't expand it
+		// as a string, instead return the value
+		// FIX: this is a hacky way of returning the value rather than
+		//      expanding the template
+		re := regexp.MustCompile(`{{ \.HProps\.(.*) }}`)
```

Comment:
IDK if it matters, but should we make this re a var/const since it doesn't change?

Reply:
Done